src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -1154,19 +1154,24 @@ private String execDockerCmd(String command, int timeout) {
     }
 
     // Suggest a performance improvement if docker build takes too long.
-    static final long DOCKER_BUILD_SOFT_TIMEOUT = 30000;
-    private void dockerIgnoreHelpMessage(long startTime) {
+    private static final long DOCKER_BUILD_SOFT_TIMEOUT = 30000;
+    private void checkDockerIgnore(long startTime) {
         if (System.currentTimeMillis() - startTime < DOCKER_BUILD_SOFT_TIMEOUT) {
             return;
         }
         File dockerfileToUse = dockerfile != null ? dockerfile : defaultDockerfile;
         if (dockerfileToUse.exists()) {
             File dockerContext = dockerfileToUse.getParentFile();
-            debug("dockerIgnoreHelpMessage, dockerContext="+dockerContext.getAbsolutePath());
+            debug("checkDockerIgnore, dockerContext="+dockerContext.getAbsolutePath());
             File dockerIgnore = new File(dockerContext, ".dockerignore");
             if (!dockerIgnore.exists()) { // provide some advice
-                warn("The docker build command is slower than expected. You may increase performance by adding " + 
-                    "unneeded files and directories such as any Liberty runtime directories to the .dockerignore file.");
+                try {
+                    warn("The docker build command is slower than expected. You may increase performance by adding " + 
+                        "unneeded files and directories such as any Liberty runtime directories to a .dockerignore file in " +
+                        dockerContext.getCanonicalPath()+ ".");
+                } catch (IOException e) {
+                    error("Exception while retrieving the path name of the docker context directory", e);
```

Comment:
Instead of just giving an error here, perhaps use dockerContext.getAbsolutePath() to get the path if an IOException occurs and still give the actual "docker build command is slow" warning.

e.g. something like
String buildContext;
try {
buildContext = dockerContext.getCanonicalPath();
} catch (IOException e) {
buildContext = dockerContext.getAbsolutePath();
}
warn(......buildContext);

Reply:
Not a bad idea.